test/framework/easyconfig.py

```diff
@@ -1177,6 +1177,28 @@ def test_dump(self):
             # parse result again
             dumped_ec = EasyConfig(test_ec)
 
+    def test_dump_extra(self):
+        """Test EasyConfig's dump() method for files containing extra values"""
+
+        rawtxt = '\n'.join([
+            "easyblock = 'EB_foo'",
+            "name = 'foo'",
+            "version = '0.0.1'",
+            "description = 'foo description'",
+            "homepage = 'http://foo.com/'",
+            "toolchain = {'name': 'dummy', 'version': 'dummy'}",
+            "foo_extra1 = 'foobar'",
+        ])
+
+        testec = os.path.join(self.test_prefix, 'test.eb')
+
+        ec = EasyConfig(None, rawtxt=rawtxt)
+        ec.dump(testec)
+        extra_regex = re.compile(r'^foo_extra1 = "foobar"', re.M)
+        self.assertTrue(extra_regex.search(read_file(testec)))
```

Comment:
maybe here you can do an exact comparison between rawtxt and what got dumped?

you'll need to make sure that rawtxt includes the necessary empty lines (and you may have to fiddle with single vs double quotes)